airgun/views/ansible_role.py

```diff
@@ -0,0 +1,75 @@
+from widgetastic.widget import Checkbox
+from widgetastic.widget import Text
+from widgetastic_patternfly import BreadCrumb
+from widgetastic_patternfly4 import Button
+from widgetastic_patternfly4 import PatternflyTable
+
+from airgun.views.common import BaseLoggedInView
+from airgun.views.common import SatTable
+from airgun.views.common import SearchableViewMixin
+from airgun.widgets import ActionsDropdown
+from airgun.widgets import Pagination
+
+
+class ImportPagination(Pagination):
+    PER_PAGE_BUTTON_DROPDOWN = ".//div[button[@id='pagination-options-menu-toggle-2']]"
+
+
+class AnsibleRolesView(BaseLoggedInView, SearchableViewMixin):
+    """Main Ansible Roles view. Prior to importing any roles, only the import_button
+    is present, without the search widget or table.
+    """
+
+    title = Text("//h1[contains(., text()='Ansible Roles')")
+    import_button = Text("//a[contains(@href, '/ansible_roles/import')]")
+    submit = Button('Submit')
+    total_imported_roles = Text("//span[@class='pagination-pf-items-total']")
+    table = SatTable(
+        './/table',
+        column_widgets={
+            'Name': Text("./a"),
+            'Hostgroups': Text("./a"),
+            'Hosts': Text("./a"),
+            'Imported at': Text("./a"),
+            'Actions': ActionsDropdown("./div[contains(@class, 'btn-group')]"),
+        },
+    )
+    pagination = Pagination()
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(
```

Comment:
Consider leveraging more of the defined widgets to assert the view:

```
return (
    self.title.is_displayed and
    self.import_button.is_displayed and
    self.table.is_displayed
)
```

Reply:
I'm making this change but leaving off `self.table.is_displayed` as the table is not present when no roles have been imported.